R/expand.R

```diff
@@ -1,92 +1,97 @@
-#' Expand data frame to include all combinations of values
+#' Expand data frame to include all possible combinations of values
 #'
-#' `expand()` is often useful in conjunction with `left_join()` if
-#' you want to convert implicit missing values to explicit missing values.
+#' `expand()` is often useful in conjunction with `right_join()` if
+#' you want to convert implicit missing values to explicit missing values
+#' (e.g., fill in gaps in your data frame).
 #' Or you can use it in conjunction with `anti_join()` to figure
-#' out which combinations are missing.
+#' out which combinations are missing (e.g., identify gaps in your
+#' data frame).
 #'
-#' `crossing()` is a wrapper around [expand_grid()] that deduplicates and sorts
-#' each input. `nesting()` is the complement to `crossing()`: it only keeps
-#' combinations of values that appear in the data.
+#' `nesting()` is a helper function for `expand` that only retains
+#' combinations of values that appear in the data.  Length-zero (empty) elements
+#' are automatically dropped.
 #'
 #' @inheritParams expand_grid
 #' @param data A data frame.
 #' @param ... Specification of columns to expand. Columns can be atomic vectors
 #'   or lists.
 #'
-#'   To find all unique combinations of x, y and z, including those not
-#'   found in the data, supply each variable as a separate argument.
-#'   To find only the combinations that occur in the data, use nest:
-#'   `expand(df, nesting(x, y, z))`.
+#' @section All possible combinations:
+#' To find all unique combinations of x, y and z, including those not
+#' present in the data, supply each variable as a separate argument.
+#' `expand(df, x, y, z)`.
 #'
-#'   You can combine the two forms. For example,
-#'   `expand(df, nesting(school_id, student_id), date)` would produce
-#'   a row for every student for each date.
+#' @section Data specific combinations:
+#' To find only the combinations that occur in the data, use `nesting`.
+#' `expand(df, nesting(x, y, z))`.
+#'
+#' @section All possible + data specific combinations:
+#' You can combine the two forms. For example,
+#' `expand(df, nesting(school_id, student_id), date)` would produce
+#' a row for each present school-student combination for all possible
+#' dates.
+#'
+#' @section `expand` with factor variables:
+#' `Expand` operates on the full set of levels defined in the factor
+#' (not just those that appear in the data).  To avoid this, use
+#' in conjunction with `forcats::fct_drop`.
```

Comment:
@hadley, I'm thinking that the description itself is useful, but that we'd skip the example to avoid adding forcats to tidyr's Suggests.

Reply:
understood and agreed!